mlir/lib/Conversion/GPUToSPIRV/GPUToSPIRVPass.cpp

```diff
@@ -71,7 +71,8 @@ void GPUToSPIRVPass::runOnOperation() {
       std::unique_ptr<ConversionTarget> target =
           spirv::getMemorySpaceToStorageClassTarget(*context);
       spirv::MemorySpaceToStorageClassMap memorySpaceMap =
-          spirv::mapMemorySpaceToVulkanStorageClass;
+          this->useOpenCL ? spirv::mapMemorySpaceToOpenCLStorageClass :
+              spirv::mapMemorySpaceToVulkanStorageClass;
```

Comment:
Can you point out which unit-test is covering this? Can you send this in a separate PR?

Reply:
Created #66445 which includes the code change and unit-test for covering.